sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryStorageStreamBundleSource.java

```diff
@@ -355,6 +351,11 @@ private boolean readNextRecord() throws IOException {
       // progress made in the current Stream gives us the overall StreamBundle progress.
       fractionOfStreamBundleConsumed =
           (currentStreamBundleIndex + fractionOfCurrentStreamConsumed) / source.streamBundle.size();
+
+      SchemaAndRecord schemaAndRecord = new SchemaAndRecord(reader.readSingleRecord(), tableSchema);
+
+      current = parseFn.apply(schemaAndRecord);
+
```

Comment:
the .apply can throw an exception. This didn't matter previously, because everything would be retried, but now we want to sometimes continue even through an exception. This means we need to advance the state of the reader fraction.